include/clang/AST/Expr.h

```diff
@@ -955,9 +955,43 @@ class DeclRefExpr final
       private llvm::TrailingObjects<DeclRefExpr, NestedNameSpecifierLoc,
                                     NamedDecl *, ASTTemplateKWAndArgsInfo,
                                     TemplateArgumentLoc> {
+public :
+  /// \brief extra information for DeclRefExprInfo.
+  class GenericFunctionCallInfo {
```

Comment:
I suggest renaming this to GenericInstInfo.  Please add a comment saying that this is information for an instantiation of a generic function.   I suggested renaming TypeNameInfo to TypeArgument.